Add debug messges when State changes

davidBar-On opened the pull request:
* Version of iperf3 (or development branch, such as `master` or
  `3.1-STABLE`) to which this pull request applies:
Master

* Issues fixed (if any): None

* Brief description of code changes (suitable for use as a commit message):
Added DEBUG_LEVEL_INFO debug messages when state changes.  I found these debug messages useful in different error cases, especially when there is a problem with the client/server interface.

bmah888 commented:
Thanks for the pull request! Just having glanced over it, this seems like a good thing to have, especially since the test state machine isn't super-well-documented and I personally regard it as a bit of magic.

davidBar-On commented:
> ... especially since the test state machine isn't super-well-documented ...

To better help with this I added a very short textual description of the states to the debug messages (and added some debug messages in additional places where the state is changed).

bmah888 commented on src/iperf_util.c:
```diff
@@ -595,3 +595,29 @@ getline(char **buf, size_t *bufsiz, FILE *fp)
+{
+    char *txt;
+
+    switch (state) {
+        case 1: txt = "Starting a Test"; break;
+        case 2: txt = "Test is Running"; break;
+        case 4: txt = "Test Ended"; break;
+        case 9: txt = "Client to Server Parameters Exchange"; break;
+        case 10: txt = "Creating Streams"; break;
+        case 11: txt = "Server Terminated"; break;
+        case 12: txt = "Client Terminated"; break;
+        case 13: txt = "Exchanging Results"; break;
+        case 14: txt = "Displaying Results"; break;
+        case 15: txt = "Starting"; break;
+        case 16: txt = "Done"; break;
+        case -1: txt = "Error"; break;
+        case -2: txt = "Serevr Error"; break;
+        default: txt = "Unknown State";
+    }
```
I think this will be really helpful in localizing issues related to the control channel or test control.

Instead of numeric constants in this switch statement, what about using the definitions such as `TEST_START`, `TEST_RUNNING`, etc. from `iperf_api.h`?

I'm wondering if the text strings should just be the names of those constants, e.g. `"TEST_START"` but I don't feel super-strongly about that.

Finally in `case -2`, the text string should be `"Server Error"`.

davidBar-On replied:
> Instead of numeric constants in this switch statement, what about using the definitions such as TEST_START, TEST_RUNNING, etc. from iperf_api.h?

Oops! Not sure why I used the numeric values...  Changed.

> I'm wondering if the text strings should just be the names of those constants, e.g. "TEST_START" but I don't feel super-strongly about that.

I was also not sure about the right approach.  I now changed that to a mixed text - using the names for all the values with added text where the name is not clear enough.  Also, added "Test reset" for value 0.

In addition, to reduce number of debug print statements, when possible I changed the `test->state = <new state>` to `iperf_set_test_state(test, <new state>)` function call and added the debug message in that function.  Also added debug messages before reading a new state.

bmah888 replied:
Thanks for the improvements, I liked the slight refactoring in `iperf_set_test_state()`.

bmah888 reviewed (changes requested):
Thanks for the PR! I think this will be very helpful for developers.

bmah888 reviewed (approved):
+1 to merge this
